### PR TITLE
Finished option provider, GUI validation/config import name fill/search bar/clear button

### PR DESCRIPTION
**Pull request by s92025592025**
Provides options for types having fixed set of input. Note type name with `Set` in the back allows multiple options to be chosen

### CODE REVIEW

File: src/main/java/org/infernus/idea/checkstyle/util/CheckStyleRuleProvider.java [hunk 1]
```diff
@@ -180,8 +219,22 @@ public ConfigRule getDefaultRuleByName(String name) {
   }
 
   /**
+<<<<<<< HEAD
```
Comment (OfekI) [inline src/main/java/org/infernus/idea/checkstyle/util/CheckStyleRuleProvider.java]: This seems like the result of a merge conflict...
It should not be in the PR

**Review by OfekI — CHANGES_REQUESTED** [review]
I left a comment.
How do you see the view interacting with this stuff?

**Comment by s92025592025** [comment]
> I left a comment.
> How do you see the view interacting with this stuff?

What I see is that the GUI or control will ask for a list of set value for a specific type, if null is returned, then it means there are no set options and user should manually type in the value.

 Note for `Access Modifier Set`, user are allowed to pick multiple options, so whether or not make that a drop-down list is up to you

**Review by OfekI — APPROVED** [review]

File: src/main/java/org/infernus/idea/checkstyle/util/PropertyValueValidator.java [hunk 2]
```diff
@@ -11,86 +15,101 @@
...
+        }
+
+        if (Pattern.matches("^subset of tokens.+$", type) && value.length() > 0) {
+            // types with "subset of tokens..." since those have no pattern we have to hard code things
+            String tokenString = type.replaceAll("subset of tokens", "").replace(".", "").trim();
+            String tokens[] = tokenString.split(", ");
+
+            if (tokens.length == 1 && tokens[0].equals("TokenTypes")) {
+                // another special case in this case
+                String tokentypes[] = {
```
Comment (OfekI) [inline src/main/java/org/infernus/idea/checkstyle/util/PropertyValueValidator.java]: Maybe we can make this a static field and not have to re-generate it every method call?

**Pull request merged**